datafusion/core/tests/sql/expr.rs

```diff
@@ -1203,121 +1203,54 @@ async fn nested_subquery() -> Result<()> {
 }
 
 #[tokio::test]
-async fn comparisons_with_null() -> Result<()> {
+async fn comparisons_with_null_lt() {
     let ctx = SessionContext::new();
-    // 1. Numeric comparison with NULL
-    let sql = "select column1 < NULL from (VALUES (1, 'foo' ,2.3), (2, 'bar', 5.4)) as t";
-    let actual = execute_to_batches(&ctx, sql).await;
-    let expected = vec![
-        "+-------------------+",
-        "| t.column1 Lt NULL |",
-        "+-------------------+",
-        "|                   |",
-        "|                   |",
-        "+-------------------+",
-    ];
-    assert_batches_eq!(expected, &actual);
 
-    let sql =
-        "select column1 <= NULL from (VALUES (1, 'foo' ,2.3), (2, 'bar', 5.4)) as t";
-    let actual = execute_to_batches(&ctx, sql).await;
-    let expected = vec![
-        "+---------------------+",
-        "| t.column1 LtEq NULL |",
-        "+---------------------+",
-        "|                     |",
-        "|                     |",
-        "+---------------------+",
+    // we expect all the following queries to yield a two null values
+    let cases = vec![
+        // 1. Numeric comparison with NULL
```

Comment:
I wanted to test a bunch of different combinations, so I rewrote the test so I could check them all